src/Analyzers/CSharp/Tests/UseImplicitObjectCreation/CSharpUseImplicitObjectCreationTests.cs

```diff
@@ -662,6 +662,68 @@ void M()
     {
         List<(int SomeName, int SomeOtherName, int YetAnotherName)> list = new();
     }
+}",
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitObjectCreation)]
+        public async Task TestWithArrayInitializerExpression()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C
+{
+    void bar()
+    {
+        var a = new C[]
+        {
+            new [|C|](),
+        };
+    }
+}",
+                FixedCode = @"
+class C
+{
+    void bar()
+    {
+        var a = new C[]
+        {
+            new(),
+        };
+    }
+}",
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitObjectCreation)]
+        public async Task TestWithCollectionInitializerExpression()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C
+{
+    void bar()
+    {
+        var a = new System.Collections.Generic.List<C>
+        {
+            new [|C|](),
+        };
+    }
+}",
+                FixedCode = @"
+class C
+{
+    void bar()
+    {
+        var a = new System.Collections.Generic.List<C>
+        {
+            new(),
+        };
+    }
 }",
                 LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
             }.RunAsync();
```

Comment:
we should have a test where someone does `new Foo<X> { new Y() }`.  e.g. where the Foo.Add method does take a Y.  In this case, we do not want this to succeed.